### ISSUE TITLE
12$: warn when destructuring in for-in loop

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1058]
The following code almost certainly has a bug because it destructures as string (because for-in loops give strings):
```
let a = [{k: 'v'}];
for (let {k} in a) {
  console.log(k);
}
let k;
for ({k} in a) {
  console.log(k);
}
```
(Same bug if destructuring an array.)

quick-lint-js should warn on this code and suggest a for-of loop instead.
